### ISSUE TITLE
Updating fileshare name and description when the status is in "creating" status

### ISSUE THREAD

**Issue by Shruthi-1MN** [opensds/opensds#810]
**Is this a BUG REPORT or FEATURE REQUEST?**:
=== RUN   TestUpdateFileShare
2019/06/04 12:27:12 
StatusCode: 202 Accepted
Response Body:
{"id":"0ed91af2-2451-4222-8bc3-e139b4e4e931","createdAt":"2019-06-04T12:27:12","updatedAt":"2019-06-04T12:27:12","tenantId":"e93b4c0934da416eb9c8d120c5d04d96","name":"test","description":"This is a test","size":1,"availabilityZone":"default","status":"creating","profileId":"d1ac0d2e-0cb3-431e-925d-e0070986f695"}
2019/06/04 12:27:12 
StatusCode: 200 OK
Response Body:
{"id":"0ed91af2-2451-4222-8bc3-e139b4e4e931","createdAt":"2019-06-04T12:27:12","updatedAt":"2019-06-04T12:27:12","tenantId":"e93b4c0934da416eb9c8d120c5d04d96","name":"test","description":"This is a test","size":1,"availabilityZone":"default","status":"available","poolId":"981eed4a-be16-5dac-b529-51148754ba5d","profileId":"d1ac0d2e-0cb3-431e-925d-e0070986f695","exportLocations":["10.20.3.73:/var/test"],"metadata":{"lvPath":"/dev/opensds-files-default/test","nfsFileshareID":"0ed91af2-2451-4222-8bc3-e139b4e4e931","nfsFileshareName":"test"}}
2019/06/04 12:27:12 
StatusCode: 200 OK
Response Body:
{"id":"0ed91af2-2451-4222-8bc3-e139b4e4e931","createdAt":"2019-06-04T12:27:12","updatedAt":"2019-06-04T12:27:12","tenantId":"e93b4c0934da416eb9c8d120c5d04d96","name":"UpdateFileShareName","description":"Update FileShare Description","size":1,"availabilityZone":"default","profileId":"d1ac0d2e-0cb3-431e-925d-e0070986f695","exportLocations":["10.20.3.73:/var/test"],"metadata":{"lvPath":"/dev/opensds-files-default/test","nfsFileshareID":"0ed91af2-2451-4222-8bc3-e139b4e4e931","nfsFileshareName":"test"}}

> Uncomment only one, leave it on its own line: 
>
> /kind bug
> /kind feature

**What happened**:

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Hotpot(release/branch) version:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

**Comment by skdwriting** [comment]
@Shruthi-1MN , please mention and provide details on what is expected and how to reproduce..

**Comment by Shruthi-1MN** [comment]
Tested with latest master branch, it updating file share name and description, only when status is available.
```
root@shruthi-VirtualBox:~/go/src/github.com/opensds/opensds# curl --header "Content-Type: application/json" --data '{"name": "31test", "description":"test fileshare", "size":2}' http://127.0.0.1:50040/v1beta/e93b4c0934da416eb9c8d120c5d04d96/file/shares
{"id":"4a845d56-762c-4f7b-bd42-5a280ac907ea","createdAt":"2019-06-12T11:26:32","updatedAt":"2019-06-12T11:26:32","tenantId":"e93b4c0934da416eb9c8d120c5d04d96","name":"31test","description":"test fileshare","size":2,"availabilityZone":"default","status":"creating","profileId":"abc19154-a964-4627-b5a4-ce5313259f57"}
root@shruthi-VirtualBox:~/go/src/github.com/opensds/opensds# curl -X GET http://127.0.0.1:50040/v1beta/e93b4c0934da416eb9c8d120c5d04d96/file/shares/4a845d56-762c-4f7b-bd42-5a280ac907ea
{"id":"4a845d56-762c-4f7b-bd42-5a280ac907ea","createdAt":"2019-06-12T11:26:32","updatedAt":"2019-06-12T11:26:33","tenantId":"e93b4c0934da416eb9c8d120c5d04d96","name":"31test","description":"test fileshare","size":2,"availabilityZone":"default","status":"available","poolId":"ce8cf5ee-d6a4-529c-8f31-a38fce95105b","profileId":"abc19154-a964-4627-b5a4-ce5313259f57","exportLocations":["100.64.41.152:/var/31test"],"metadata":{"lvPath":"/dev/opensds-files-default/31test","nfsFileshareID":"4a845d56-762c-4f7b-bd42-5a280ac907ea","nfsFileshareName":"31test"}}
root@shruthi-VirtualBox:~/go/src/github.com/opensds/opensds# curl -X PUT --header "Content-Type: application/json" --data '{"name":"test11", "description":"sample test"}' http://127.0.0.1:50040/v1beta/e93b4c0934da416eb9c8d120c5d04d96/file/shares/4a845d56-762c-4f7b-bd42-5a280ac907ea
{"id":"4a845d56-762c-4f7b-bd42-5a280ac907ea","createdAt":"2019-06-12T11:26:32","updatedAt":"2019-06-12T11:27:59","tenantId":"e93b4c0934da416eb9c8d120c5d04d96","name":"test11","description":"sample test","size":2,"availabilityZone":"default","status":"available","poolId":"ce8cf5ee-d6a4-529c-8f31-a38fce95105b","profileId":"abc19154-a964-4627-b5a4-ce5313259f57","exportLocations":["100.64.41.152:/var/31test"],"metadata":{"lvPath":"/dev/opensds-files-default/31test","nfsFileshareID":"4a845d56-762c-4f7b-bd42-5a280ac907ea","nfsFileshareName":"31test"}}
```

**Comment by Shruthi-1MN** [comment]
You can close it.
